modules/social_features/social_group/modules/social_group_welcome_message/social_group_welcome_message.module

```diff
@@ -0,0 +1,150 @@
+<?php
+
+/**
+ * @file
+ * Primary module hooks for social_group_welcome_message module.
+ */
+
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\group\Entity\GroupContentInterface;
+
+/**
+ * Implements hook_entity_base_field_info().
+ */
+function social_group_welcome_message_entity_base_field_info(EntityTypeInterface $entity_type) {
+  $fields = [];
+
+  // Add a 'Welcome message' base fields to all group types.
+  if ($entity_type->id() === 'group') {
+    $fields['private_message_send'] = BaseFieldDefinition::create('boolean')
+      ->setLabel(t('Send a private message people joined'))
+      ->setRevisionable(TRUE)
+      ->setTranslatable(TRUE)
+      ->setDefaultValue(FALSE)
+      ->setDisplayOptions('form', [
+        'type' => 'boolean_checkbox',
+        'settings' => [
+          'display_label' => TRUE,
+        ],
+        'weight' => 99,
+      ])
+      ->setDisplayConfigurable('form', TRUE);
+
+    $fields['private_message_body'] = BaseFieldDefinition::create('text_long')
+      ->setLabel(t('Message'))
+      ->setRevisionable(TRUE)
+      ->setTranslatable(TRUE)
+      ->setDisplayOptions('form', [
+        'type' => 'text_textfield',
+        'weight' => 100,
+      ])
+      ->setDisplayConfigurable('form', TRUE);
+  }
+
+  return $fields;
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function social_group_welcome_message_form_alter(array &$form, FormStateInterface $form_state, $form_id) {
+  $social_group_types = [
+    'open_group',
+    'closed_group',
+    'public_group',
+  ];
+
+  \Drupal::moduleHandler()->alter('social_group_types', $social_group_types);
```

Comment:
Do we have a default implementation for this already? It feels a bit weird to define the default group types and alter them.

We should probably just have a service that caches this per request instead.

Reply:
Good point but it should be a general fix of approach getting all social group types because a lot of modules use the the same approach as I implemented here. Thats why I think if we agree to improve this it shouldn't be a part of this story.

Reply:
Given the time constraints, adding this as a follow-up on Drupal.org is fine. I think in the future it would be acceptable to include such improvements in a pull request. Following the girl-/boy-scouts motto of "Leave the code better than you found it" :D 

Of course this always depends on how big the required changes are ☺️